pkg/compact/schedule_delete.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package compact
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/block"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	"github.com/thanos-io/thanos/pkg/runutil"
+)
+
+// ScheduleDeleteFilename is file to store compactor metadata
+// about when the block is scheduled to be deleted.
+const ScheduleDeleteFilename = "compactor-meta.json"
+
+// CompactorMeta stores block id and when block was marked for deletion.
+type CompactorMeta struct {
+	ID           ulid.ULID `json:"id"`
```

Comment:
Also, let's please add some field like `Version` here so that we could differentiate between different versions in the future if it will change :P something similar to what we have in the meta file.

Reply:
I will work on adding this.

Reply:
Personally I'm not a fan of versioning JSON structures. They are already easy enough to extend, or to check for missing fields.

Reply:
I disagree with this - it's not proto (: And we already version Meta, so let's stick to this.

Reply:
> I disagree with this - it's not proto (: And we already version Meta, so let's stick to this.. please =D

Just to be clear — you disagree with my opposition to having Version, right? It seems pointless to me to do it in JSON, but if Thanos uses this pattern elsewhere, it’s fine with me.